plugins/Referrers/Columns/Keyword.php

```diff
@@ -36,6 +36,18 @@ public function onNewVisit(Request $request, Visitor $visitor, $action)
         return $information['referer_keyword'];
     }
 
+    public function onExistingVisit(Request $request, Visitor $visitor, $action)
+    {
+        $information = $this->getReferrerInformationFromRequest($request, $visitor);
+        if ($this->isCurrentReferrerDirectEntry($visitor)
+            && $information['referer_type'] == Common::REFERRER_TYPE_CAMPAIGN
```

Comment:
As well as Campaigns, expecting also Search engine / Website / Social to overwrite a "Direct entry" referrer (not just campaigns)

Reply:
updated